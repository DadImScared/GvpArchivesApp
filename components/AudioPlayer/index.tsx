import * as React from "react";
import { StyleSheet, View } from "react-native";

import { withAudioPlayer } from "../withAudioPlayer";
import ButtonGroup from "./ButtonGroup";
import DurationSeeker from "./DurationSeeker";
import { IAudioPlayerState } from "./Provider";
import Title from "./Title";

interface IProps {
  audio: IAudioPlayerState;
}

const styles = StyleSheet.create({
  wrapper: {
    backgroundColor: "white",
    bottom: 0,
    padding: 8,
    position: "absolute",
    width: "100%",
  }
});

export class AudioPlayer extends React.Component<IProps> {
  render() {
    const {
      audio: {
        isSeeking,
        isPlaying,
        onPlayPausePressed,
        getSongName,
        getSeekSliderPosition,
        getTimeStamp,
        onSeekSliderChange,
        onSeekSliderComplete

      }
    } = this.props;
    return (
      <View style={styles.wrapper}>
        <Title getSongName={getSongName} />
        <DurationSeeker
          isSeeking={isSeeking}
          getSeekSliderPosition={getSeekSliderPosition}
          getTimeStamp={getTimeStamp}
          onSeekSliderChange={onSeekSliderChange}
          onSeekSliderComplete={onSeekSliderComplete}
        />
        <ButtonGroup
          onPlayPausePressed={onPlayPausePressed}
          isPlaying={isPlaying}
        />
      </View>
    );
  }
}

export default withAudioPlayer(AudioPlayer) as any; // property error without "any"
