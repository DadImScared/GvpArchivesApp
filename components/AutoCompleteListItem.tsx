import * as React from "react";
import { Text } from "react-native";

import { ListItem } from "native-base";

interface IProps {
  text: string;
}

export class AutoCompleteListItem extends React.Component<IProps> {

  render() {
    return (
      <ListItem button={true} noIndent={true}>
        <Text>{this.props.text}</Text>
      </ListItem>
    );
  }
}

export default AutoCompleteListItem;
