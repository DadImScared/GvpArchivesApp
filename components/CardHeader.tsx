import * as React from "react";

import { Body, CardItem, Text } from "native-base";
import { routes } from "./SideBar";

interface IProps {
  item: any;
  extraSubHeader?: string;
}

export const CardHeader: React.FunctionComponent<IProps> = ({ item, extraSubHeader }) => (
  <CardItem header={true}>
    <Body>
      <Text>{item.title.trim()}</Text>
      <Text note={true}>{routes[item.category]}{extraSubHeader ? `${extraSubHeader}` : null}</Text>
    </Body>
  </CardItem>
);

export default CardHeader;
