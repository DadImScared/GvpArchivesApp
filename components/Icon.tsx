import * as React from "react";

import { Icon as NativeBaseIcon, NativeBase } from "native-base";

export class Icon extends React.Component<NativeBase.Icon> {
  render() {
    return (
      <NativeBaseIcon {...this.props} />
    );
  }
}

export default Icon;
