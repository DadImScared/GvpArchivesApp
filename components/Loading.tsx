import * as React from "react";

import { Spinner } from "native-base";
import { View } from "react-native";

export const Loading: React.FunctionComponent<{ isLoading: boolean }> = ({ isLoading }) => isLoading ? (
  <View>
    <Spinner />
  </View>
) : null;

export default Loading;
