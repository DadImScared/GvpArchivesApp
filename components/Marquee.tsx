import * as React from "react";

import Ticker, { ITextMarqueeProps } from "react-native-text-ticker";

const ForwardedMarquee = React.forwardRef<Ticker, ITextMarqueeProps>(({ children, ...other }, ref) => (
  <Ticker {...other} ref={ref}>
    {children}
  </Ticker>
));

export default ForwardedMarquee;
