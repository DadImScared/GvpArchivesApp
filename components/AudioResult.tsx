import * as React from "react";

import { Button, Card, CardItem, Left } from "native-base";
import { StyleSheet, Text } from "react-native";

import { Item } from "../reducers/itemsById";
import CardHeader from "./CardHeader";

export interface IAudioResultProps {
  item: Item;
  [key: string]: any;
}

const styles: any = StyleSheet.create({
  button: {
    alignItems: "center",
    justifyContent: "center",
    margin: 4,
    width: "30%"
  }
});

export const AudioResult: React.FC<IAudioResultProps> = ({ item }) => (
  <Card>
    <CardHeader item={item} />
    <CardItem footer={true}>
      <Left>
        <Button primary={true} style={styles.button}><Text style={{ color: "white" }}>Play</Text></Button>
        <Button info={true} style={styles.button}><Text>Playlist</Text></Button>
      </Left>
    </CardItem>
  </Card>
);

export default AudioResult;
