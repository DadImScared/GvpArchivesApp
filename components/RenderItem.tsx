import * as React from "react";

import { Card } from "native-base";
import { View } from "react-native";

import { Item } from "../reducers/itemsById";
import AudioResult from "./AudioResult";
import CardHeader from "./CardHeader";

const CATEGORIES: { [key: string]: any | string } = {
  bhagavatpatrika: "Bhagavat Patrika",
  book: "Books",
  harikatha: "Harikatha",
  harmonistmagazine: "Harmonist Magazine",
  harmonistmonthly: "Harmonist Monthly",
  lecture: "Lecture",
  movie: "Movies",
  song: AudioResult,
};

export interface IRenderItemProps {
  item: Item;
}

export const RenderItem: React.FunctionComponent<IRenderItemProps> = ({ item }) => (
  <View style={{ padding: 4 }}>
    {
      typeof CATEGORIES[item.category] === "string" ?
        <Card>
          <CardHeader item={item} />
        </Card>
        :
        React.createElement(CATEGORIES[item.category], { item })}
  </View>
);

export default RenderItem;
