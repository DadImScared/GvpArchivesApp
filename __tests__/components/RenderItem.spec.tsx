import * as React from "react";
import * as renderer from "react-test-renderer";

import { IRenderItemProps, RenderItem } from "../../components/RenderItem";

let props: IRenderItemProps;

describe("RenderItem", () => {
  beforeEach(() => {
    props = {
      item: {
        category: "song",
        item_id: "1",
        link: "itemlink",
        title: "itemtitle"
      }
    };
  });

  it("should render default component", () => {
    props = {
      item: {
        ...props.item,
        category: "harmonistmonthly"
      }
    };
    const tree = renderer.create(<RenderItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render custom component", () => {
    const tree = renderer.create(<RenderItem {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
