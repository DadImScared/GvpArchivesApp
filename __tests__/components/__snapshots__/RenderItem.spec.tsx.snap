// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderItem should render custom component 1`] = `
<View
  style={
    Object {
      "padding": 4,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      header={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": null,
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
          itemtitle
        </Text>
        <Text
          note={true}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "200",
              "marginRight": 20,
            }
          }
          uppercase={false}
        >
          Songs
        </Text>
      </View>
    </View>
    <View
      footer={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": null,
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 10,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 10,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#007aff",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "margin": 4,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
              "width": "30%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
              }
            }
          >
            Play
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#62B1F6",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "margin": 4,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
              "width": "30%",
            }
          }
        >
          <Text>
            Playlist
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`RenderItem should render default component 1`] = `
<View
  style={
    Object {
      "padding": 4,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      }
    }
  >
    <View
      header={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "borderBottomWidth": null,
          "borderRadius": 2,
          "flexDirection": "row",
          "paddingBottom": 15,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 15,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "stretch",
            "flex": 1,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            }
          }
          uppercase={false}
        >
          itemtitle
        </Text>
        <Text
          note={true}
          style={
            Object {
              "color": "#808080",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "200",
              "marginRight": 20,
            }
          }
          uppercase={false}
        >
          Harmonist Monthly
        </Text>
      </View>
    </View>
  </View>
</View>
`;
